src/components/Modal/ProfileModal.tsx

```diff
@@ -8,8 +8,40 @@ export class ProfileModal extends React.Component<{
   close: Function;
   user: firebase.User;
   userImage: string | null;
+  isSubscriber?: boolean;
 }> {
-  public state = { resetDisabled: false, deleteConfirmOpen: false };
+  public state = {
+    resetDisabled: false,
+    deleteConfirmOpen: false,
+    discordOpen: false,
+    discordUsername: '',
+    discordDiscriminator: '',
+    fetchedDiscordUsername: '',
+    fetchedDiscordDiscriminator: '',
+    discordError: '',
+  };
+
+  openDiscord = async () => {
+    await this.getDiscord();
+    this.setState({ discordOpen: true });
+  };
+
+  closeDiscord = async () => {
+    this.setState({
+      discordOpen: false,
+      discordUsername: '',
+      discordDiscriminator: '',
+      discordError: '',
+    });
+  };
+
+  setDiscordUsername(value: string) {
```

Comment:
rather than making the user enter a name and risk errors (or entering someone else's name), could we integrate with Discord OAuth to automatically fetch the user's info?